facades/TeraEd/src/main/java/org/terasology/editor/input/AwtMouseDevice.java

```diff
@@ -0,0 +1,159 @@
+    private double yPosDelta;
+
+    public AwtMouseDevice(RenderingConfig renderingConfig) {
+        this.renderingConfig = renderingConfig;
+        this.uiScale = renderingConfig.getUiScale() / 100f;
+        renderingConfig.subscribe(RenderingConfig.UI_SCALE, this);
+    }
+
+    public void registerToAwtGlCanvas(AWTGLCanvas canvas) {
+        canvas.addMouseListener(new MouseListener() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+            }
+
+            @Override
+            public void mousePressed(MouseEvent e) {
+                int button = e.getButton() - 1;
+                buttonStates.add(button);
+                MouseInput mouseInput = MouseInput.find(InputType.MOUSE_BUTTON, button);
+                queue.offer(new MouseAction(mouseInput, ButtonState.DOWN, getPosition()));
+            }
+
+            @Override
+            public void mouseReleased(MouseEvent e) {
+                int button = e.getButton() - 1;
+                buttonStates.remove(button);
+                MouseInput mouseInput = MouseInput.find(InputType.MOUSE_BUTTON, button);
+                queue.offer(new MouseAction(mouseInput, ButtonState.UP, getPosition()));
+            }
+
+            @Override
+            public void mouseEntered(MouseEvent e) {
+
+            }
+
+            @Override
+            public void mouseExited(MouseEvent e) {
+
+            }
+        });
+        canvas.addMouseMotionListener(new MouseMotionListener() {
+            @Override
+            public void mouseDragged(MouseEvent e) {
+                updateMouse(e.getX(), e.getY());
+            }
+
+            @Override
+            public void mouseMoved(MouseEvent e) {
+                updateMouse(e.getX(), e.getY());
+            }
+        });
+
+        canvas.addMouseWheelListener(e -> {
+            int yOffset = e.getUnitsToScroll();
+            if (yOffset != 0.0) {
+                int id = (yOffset > 0) ? 1 : -1;
+                queue.offer(new MouseAction(InputType.MOUSE_WHEEL.getInput(id), 1, getPosition()));
+            }
+        });
+    }
+
+    @Override
+    public void update() {
+    }
+
+    private void updateMouse(double x, double y) {
+        xPosDelta = x - this.xPos;
+        yPosDelta = y - this.yPos;
+        this.xPos = x;
+        this.yPos = y;
+    }
+
+    @Override
+    public Vector2i getPosition() {
+        return new Vector2i((int) (xPos / this.uiScale), (int) (yPos / this.uiScale));
+    }
+
+    @Override
+    public Vector2d getDelta() {
+
+        Vector2d result = new Vector2d(xPosDelta, yPosDelta);
+        return result;
+    }
+
+    @Override
+    public boolean isButtonDown(int button) {
+        return buttonStates.contains(button);
+    }
+
+    @Override
+    public boolean isVisible() {
+        return !mouseGrabbed;
+    }
+
+    @Override
+    public void setGrabbed(boolean newGrabbed) {
+        if (newGrabbed != mouseGrabbed) {
+            mouseGrabbed = newGrabbed;
```

Comment:
see:
https://stackoverflow.com/questions/32159065/how-to-grab-a-mouse-in-java-swing

Reply:
This will require additional logic for shifting between regular and grab mode. There is also issue with rotating already.